src/main/java/com/hotels/mutantswarm/MutantSwarmExtension.java

```diff
@@ -0,0 +1,212 @@
+import com.klarna.hiverunner.HiveShellContainer;
+import com.klarna.hiverunner.annotations.HiveSQL;
+import com.klarna.hiverunner.builder.Script;
+import com.klarna.hiverunner.config.HiveRunnerConfig;
+import com.klarna.reflection.ReflectionUtils;
+import com.klarna.hiverunner.builder.HiveRunnerScript;
+
+public class MutantSwarmExtension extends HiveRunnerExtension implements AfterAllCallback, TestWatcher,
+    TestTemplateInvocationContextProvider, TestInstancePostProcessor, AfterEachCallback {
+
+  private static final Logger log = LoggerFactory.getLogger(MutantSwarmExtension.class);
+  private static AtomicReference<ExecutionContext> contextRef = new AtomicReference<>();
+
+  public HiveShellContainer container;
+  private HiveRunnerConfig HiveRunnerConfig = new HiveRunnerConfig();
+  private int testNumber = -1;
+  private List<Mutant> mutants = new ArrayList<Mutant>();
+  private List<MutatedSource> mutatedSources = new ArrayList<MutatedSource>();
+  private ExecutionContext results = contextRef.get();
+  private boolean firstTestPassed = true;
+  private MutantSwarmCore core = new MutantSwarmCore();
+
+  public MutantSwarmExtension() {}
+
+  @Override
+  public boolean supportsTestTemplate(ExtensionContext context) {
+    if (!context.getTestMethod().isPresent()) {
+      return false;
+    }
+    return true;
+  }
+
+  @Override
+  public Stream<TestTemplateInvocationContext> provideTestTemplateInvocationContexts(ExtensionContext context) {
+    testNumber = -1;
+    firstTestPassed = true;
+
+    setFirstScripts(context);
+
+    if (contextRef.get() == null) {
+      Swarm swarm = core.generateSwarm(scriptsUnderTest, log, HiveRunnerConfig.getCommandShellEmulator());
+      SwarmResultsBuilder swarmResultBuilder = new SwarmResultsBuilder(swarm,
+          context.getRequiredTestClass().toString());
+      contextRef.compareAndSet(null, new ExecutionContext(swarm, swarmResultBuilder));
+    }
+
+    mutants = contextRef.get().swarm.getMutants();
+    ExecutionContext executionContext = contextRef.get();
+    for (Mutant mutant : contextRef.get().swarm.getMutants()) {
+      MutatedSource mutatedSource = core.mutateSource(executionContext.getSource(), mutant);
+      mutatedSources.add(mutatedSource);
+    }
+
+    results = contextRef.get();
+
+    // Here we specify the number of times a test should be repeated, but the scripts will be mutated accordingly for
+    // each individual test in the postProcessTestInstance method
+    return IntStream.rangeClosed(1, mutants.size() + 1).mapToObj(repitition -> new MutantSwarmTestTemplate());
+  }
+
+  @Override
+  public void postProcessTestInstance(Object target, ExtensionContext extensionContext) {
+    testNumber++;
+
+    // The first test per swarm should run normally and without mutations
+    if (testNumber != 0) {
+      scriptsUnderTest.clear();
+      MutatedSource mutatedSource = mutatedSources.get(testNumber - 1);
+      List<MutantSwarmScript> MScripts = mutatedSource.getScripts();
```

Comment:
```suggestion
      List<MutantSwarmScript> mutatedScripts= mutatedSource.getScripts();
```
We only use upper case in first character for class names.